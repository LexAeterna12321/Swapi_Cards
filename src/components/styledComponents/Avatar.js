import styled from "styled-components";
const Avatar = styled.section`
  width: 90%;
  margin: 0 auto;
`;

export default Avatar;
