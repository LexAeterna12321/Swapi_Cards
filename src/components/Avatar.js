import React from "react";
import PropTypes from "prop-types";
import { Avatar as StyledAvatar, Image } from "./styledComponents";

const Avatar = ({ name }) => (
  <StyledAvatar>
    {name === "C-3PO" ? (
      <Image src="./img/c3po.jpg" alt="c-3po" />
    ) : (
      <Image src="./img/luke.jpg" alt="Luke Skywalkwer" />
    )}
  </StyledAvatar>
);

Avatar.propTypes = {
  name: PropTypes.string.isRequired
};

export default Avatar;
