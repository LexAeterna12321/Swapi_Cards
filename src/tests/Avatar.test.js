import React from "react";
import Avatar from "../components/Avatar";
import { shallow } from "enzyme";
it("expects to render SwapiCharCard component", () => {
  expect(shallow(<Avatar />)).toMatchSnapshot();
  expect(
    shallow(<Avatar name={"C-3PO"} />).contains(
      <img className="card__avatar__img" src="./img/c3po.jpg" alt="c-3po" />
    )
  ).toEqual(true);
  expect(
    shallow(<Avatar name={"Luke Skywalker"} />).contains(
      <img
        className="card__avatar__img"
        src="./img/luke.jpg"
        alt="Luke Skywalkwer"
      />
    )
  ).toEqual(true);
});
