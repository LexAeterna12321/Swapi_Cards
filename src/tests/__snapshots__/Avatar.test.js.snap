// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render SwapiCharCard component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Avatar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="Luke Skywalkwer"
        className="card__avatar__img"
        src="./img/luke.jpg"
      />,
      "className": "card__avatar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "Luke Skywalkwer",
        "className": "card__avatar__img",
        "src": "./img/luke.jpg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="Luke Skywalkwer"
          className="card__avatar__img"
          src="./img/luke.jpg"
        />,
        "className": "card__avatar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Luke Skywalkwer",
          "className": "card__avatar__img",
          "src": "./img/luke.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
